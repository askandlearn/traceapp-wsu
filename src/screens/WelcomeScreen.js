import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  TouchableOpacity,
} from 'react-native';

//Create the Welcome Page
const WelcomeScreen = (props) => {
  return (
    <View style={styles.container}>
      <ImageBackground
        blurRadius={3}
        style={styles.backgroundImage}
        source={require('../images/Trace-3D.png')}>
        <TouchableOpacity
          style={styles.button}
          onPress={() => props.navigation.navigate('Login')}>
          <Text style={styles.buttonText}>SIGN IN</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => props.navigation.navigate('SignUp')}>
          <Text style={styles.buttonText}>SIGN UP</Text>
        </TouchableOpacity>
      </ImageBackground>
    </View>
  );
};

/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
              STYLE SHEET
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  backgroundImage: {
    flex: 1,
    resizeMode: 'cover',
    justifyContent: 'center',
  },
  button: {
    alignItems: 'center',
    marginHorizontal: '10%',
    marginVertical: '5%',
    padding: 10,
    borderRadius: 20,
    backgroundColor: '#ff0000',
    shadowColor: '#000000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },
  buttonText: {
    color: '#FFFFFF',
    fontWeight: 'bold',
  },
});

export default WelcomeScreen;
