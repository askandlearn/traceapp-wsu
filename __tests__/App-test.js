/**
 * @format
 */

import 'react-native';
import React from 'react';
import WelcomeScreen from '../src/screens/WelcomeScreen'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const welcome = renderer.create(<WelcomeScreen/>)

test('snapshot', () => {
  expect(welcome).toMatchSnapshot();
});
