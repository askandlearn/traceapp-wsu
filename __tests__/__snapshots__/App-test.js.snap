// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      blurRadius={3}
      source={
        Object {
          "testUri": "../../../src/images/Trace-3D.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff0000",
          "borderRadius": 20,
          "elevation": 1,
          "marginHorizontal": "10%",
          "marginVertical": "5%",
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontWeight": "bold",
          }
        }
      >
        SIGN IN
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ff0000",
          "borderRadius": 20,
          "elevation": 1,
          "marginHorizontal": "10%",
          "marginVertical": "5%",
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontWeight": "bold",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;
